src/utils/url.ts

```diff
@@ -105,6 +105,21 @@ export const isValidUrl = (url: string): boolean => {
   }
 };
 
+export const isStringAValidUrl = (urlString: string) => {
+  if (!urlString) return false;
+
+  try {
+    const url = new URL(urlString);
+
+    return VALID_URL_PROTOCOLS.includes(url.protocol);
+  } catch (e) {
+    if (e instanceof Error) {
+      return false;
+    }
+    throw e;
+  }
+};
+
```

Comment:
This is identical to the function `isValidUrl` above except for the first line. And for the `instanceof` check.
As this is used in one place you can simply check `if (!urlString) return false;` for that in that one place and use `isValidUrl` there. Also please replace `e.name === "TypeError" with `instanceof` check inside `idValidUrl`.

Reply:
fixed by removing new foo and refactoring the `isValidUrl` d04b6de